### PR TITLE
Add pre-install, post-install hooks to install-ptfe.sh

### PR DESCRIPTION
**Pull request by aaron-lane**
## Background

Primarily, this branch adds pre-install and post-install script hooks to `install-ptfe.sh`.

Secondarily, this makes some minor development changes:
- ignores variable and state files
- adds SSH agent forwarding
- pins the Terraform CLI version

[Asana task](https://app.asana.com/0/1170474914009341/1174362877426282/f)

## How Has This Been Tested

I provisioned a cluster using the root module and verified that the default scripts were executed on primaries and secondaries.

### Test Configuration

* Terraform Version: 0.12.24

## This PR makes me feel

Windows 95 CD

### CODE REVIEW

**Comment by aaron-lane** [comment]
@bnferguson Nice. I didn't consider defining separate scripts for cloud-init. Seems much cleaner. I'll update the GCP one to do the same rather than the script injection.

File: variables.tf [hunk 1]
```diff
@@ -187,6 +181,25 @@ variable "tags" {
   default     = {}
 }
 
+variable "postinstall_script" {
+  default     = <<-EOD
+  #!/bin/bash
+
+  echo 'A post-install script was not provided.'
+  EOD
```
Comment (erindatkinson) [inline variables.tf:189]: so this isn't an explicit suggestion or a blocking question, but do we as a team have an opinionated stance on heredoc format over using file() for non templated things, this script is small enough to not run into any issues, but to align expectations I'm wondering if we want to put all these in the `files/` dir as like postinstall.default.sh

Comment (aaron-lane) [reply variables.tf:189]: I think that sounds reasonable!

Comment (aaron-lane) [reply variables.tf]: How does https://github.com/hashicorp/terraform-aws-terraform-enterprise/pull/79/commits/b87a7fc48195f3d15e3b200c313c8dab951e1f11 look?

Comment (erindatkinson) [reply variables.tf]: Looks great!

**Pull request merged**